docs/artifacts/includes/nuget/nuget-publish-endpoint.md

```diff
@@ -18,14 +18,69 @@ ms.date: 02/18/2022
 
     :::image type="content" source="../../media/nuget-connect-feed.png" alt-text="NuGet.exe feed connection":::
 
-1. If this is the first time using Azure Artifacts with Nuget.exe, select **Get the tools** button and follow the instructions to install the prerequisites.
+1. When using a local development environment, it is recommended to have the [Azure Artifacts Credential Provider](https://github.com/microsoft/artifacts-credprovider) installed which provides authentication when pushing or downloading packages [see examples](#examples).
 
     1. Download the [latest NuGet version](https://www.nuget.org/downloads).
-    1. Download and install the [Azure Artifacts Credential Provider](https://github.com/microsoft/artifacts-credprovider#azure-artifacts-credential-provider).
 
-1. Follow the instructions in the **Project setup** to add a config file. 
+    1. Follow the instructions in the **Project setup** to add a config file. 
+
+        :::image type="content" source="../../media/project-setup.png" alt-text="Project setup":::
+
+1. For Azure hosted builds that interact with Azure Artifacts feeds, you must use the [NuGet Authenticate](https://docs.microsoft.com/azure/devops/pipelines/tasks/package/nuget-authenticate) task for the authentication [(see examples)](#examples).
+
+1. For Github actions builds that interact with Azure Artifacts feeds, you may use [Github setup dotnet action](https://github.com/actions/setup-dotnet) for the authentication [(see examples)](#examples).
+
+1. For other hosted builds that interact with Azure Artifacts feeds, you may use [Azure PAT]([url](https://docs.microsoft.com/en-us/azure/devops/organizations/accounts/use-personal-access-tokens-to-authenticate?view=azure-devops&tabs=Windows)) saved in `nuget.config` file for the authentication [(see examples)](#examples), but please exercise caution because it may leak your PAT as part of build environement log collection or source control checkout, artifact upload accidentally.
+
+## Examples
+
+### Pushing from a local development environment
+
+When pushing from local development environment to Azure Artifacts, it is recommended to have [Azure Artifacts Credential Provider](https://github.com/microsoft/artifacts-credprovider) installed for the authentication. Please note here the API key "AZ" is only used as a placeholder.
```

Comment:
Sorry, my current PR doesn't reflect latest changes in private branch we built for preview. I should have been more explicit.

We made some adjustments in private branch so it may look different from my current branch
Could you look at private branch instead of changes in this PR?

**nuget.exe push example**: https://review.docs.microsoft.com/en-us/azure/devops/artifacts/nuget/publish?view=azure-devops&branch=pr-en-us-4782
[Azure Artifacts Credential Provider](https://github.com/microsoft/artifacts-credprovider) requirement added into [Prerequisites section](https://review.docs.microsoft.com/en-us/azure/devops/artifacts/nuget/publish?view=azure-devops&branch=pr-en-us-4782#prerequisites).
We added push from other [external sources (for teamcity, jenkins etc ..) section](https://review.docs.microsoft.com/en-us/azure/devops/artifacts/nuget/publish?view=azure-devops&branch=pr-en-us-4782#publish-packages-from-external-sources) at bottom.

**Push from pipeline example**: We added script example as alternative to nuget.command task in pipeline example: https://review.docs.microsoft.com/en-us/azure/devops/pipelines/artifacts/nuget?toc=%2Fazure%2Fdevops%2Fartifacts%2Ftoc.json&bc=%2Fazure%2Fdevops%2Fartifacts%2Fbreadcrumb%2Ftoc.json&view=azure-devops&branch=pr-en-us-4782&tabs=yaml#publish-a-package

Reply:
Can we review that PR instead?

Reply:
I don't have access to view the preview.

Reply:
Private PR link is: https://github.com/MicrosoftDocs/azure-devops-docs-pr/pull/4782
If you get 404 error then you need to join `MicrosoftDocs` org which is pretty straight forward, [click step 3 here](https://review.docs.microsoft.com/en-us/help/get-started/setup-github?branch=main#3-join-microsoftdocs-and-other-organizations).  If you have any comment let's continue on private PR.

Actually, change is already live now https://docs.microsoft.com/en-us/azure/devops/artifacts/nuget/publish?view=azure-devops, so preview link is deleted. If you want to compare, then [previous version is here](https://review.docs.microsoft.com/en-us/azure/devops/artifacts/nuget/publish?branch=pazand-patch-1&view=azure-devops).

This PR show merged because private PR based on this, it was automatic step once private PR go live.
Let's continue discussion on private PR.

Thank you.

Reply:
That doesn't seem to mentioned the authenticate task at all, is it on purpose?

Reply:
Authenticate task example is here [Azure Pipelines (YAML/Classic)](https://docs.microsoft.com/en-us/azure/devops/pipelines/artifacts/nuget?toc=%2Fazure%2Fdevops%2Fartifacts%2Ftoc.json&bc=%2Fazure%2Fdevops%2Fartifacts%2Fbreadcrumb%2Ftoc.json&view=azure-devops&tabs=yaml#publish-a-package) since it's related only to Azire pipeline. It's how they organize their docs.